src/site/stages/build/process-cms-exports/generator-cms-content-model/generators/app/index.js

```diff
@@ -193,48 +193,96 @@ module.exports = class extends Generator {
+      if (copiedUuids.has(uuid)) return;
+      copiedUuids.add(uuid);
+
+      // Iterate through the non-blacklisted properties
+      // When an entity reference is found, copy it over and recurse on it
+      Object.keys(entity)
+        .filter(k => !propBlacklist.includes(k))
+        .forEach(propName => {
+          // Properties should always be arrays, but just in case, check
+          if (Array.isArray(entity[propName])) {
+            entity[propName].forEach((p, index) => {
+              if (p.target_type && p.target_uuid) {
+                // Found an entity reference!
+                const fileName = `${p.target_type}.${p.target_uuid}.json`;
+                const sourceFile = path.join(tomeSyncContent, fileName);
+                const destFile = path.join(
+                  processEntitiesRoot,
+                  'tests/entities/',
+                  fileName,
                 );
-              } else
-                this.log(
-                  chalk.green(
-                    `Found required child (${propName}): ${childFileName}`,
+                if (!fs.existsSync(destFile)) {
+                  try {
+                    fs.copyFileSync(sourceFile, destFile);
+                    this.log(
+                      chalk.grey(`${uuid}: `),
+                      chalk.green(
+                        `Added child entity (${propName}[${index}]): ${fileName}`,
+                      ),
+                    );
+                  } catch (e) {
+                    this.log(
+                      chalk.grey(`${uuid}: `),
+                      chalk.red(`Error copying ${fileName}.`),
+                    );
+                    this.log(
+                      chalk.grey(`${uuid}: `),
+                      chalk.yellow('  This UUID: '),
+                      entity.uuid[0].value,
+                    );
+                    this.log(
+                      chalk.grey(`${uuid}: `),
+                      chalk.yellow('  Child found at: '),
+                      `${propName}[${index}]`,
+                    );
+                    this.log(chalk.yellow('  Child: '), p);
+                    throw e;
+                  }
+                } else
+                  this.log(
+                    chalk.grey(`${uuid}: `),
+                    chalk.blue(
+                      `Found child entity (${propName}[${index}]): ${fileName}`,
+                    ),
+                  );
+
+                // Recurse!
+                copyChildren(
+                  JSON.parse(
+                    fs
+                      .readFileSync(
+                        path.join(
+                          tomeSyncContent,
+                          `${p.target_type}.${p.target_uuid}.json`,
+                        ),
+                      )
+                      .toString('utf8'),
                   ),
                 );
-            }
-          });
-        } else this.log(`${propName} is not an array. That's unexpected.`);
```

Comment:
I wonder if removing this bit of logging will cause issues with debugging later down the road when/if the structure of the data changes

Reply:
Maybe. It was logging a lot of `metatag` not being an array. :man_shrugging: